src/utils/manageLocalStorage.ts

```diff
@@ -1,22 +1,20 @@
 import { GroupInfo } from "./globalTypes";
 
-const getLocalGroups = (): GroupInfo[] | [] => {
+const getLocalGroups = (): GroupInfo[] => {
   // push local storage to the group state
-  const res: string =
-    localStorage.getItem(import.meta.env.VITE_LOCALLY_STORED_GROUPS) ||
-    "undefined";
+  const res = localStorage.getItem(import.meta.env.VITE_LOCALLY_STORED_GROUPS)
 
   // exit function if there was no local storage
-  if (res === "undefined") return [];
+  if (!res) return [];
 
   // return array of groups from local storage
   const parsedRes: GroupInfo[] = JSON.parse(res);
   return parsedRes;
 };
 
 const setLocalGroups = (groups: GroupInfo[]) => {
-  // stringify groups array
-  const stringifiedGroups: string = JSON.stringify(groups);
+  // prepare groups for storage
+  const stringifiedGroups = JSON.stringify(groups);
```

Comment:
`JSON.stringify(groups)` === `stringify groups array`

Unless there is some unwieldy complexity, you typically don't need comments that describe exactly what you're doing.  What you might want to do instead is use a comment to explain the why (i.e. groups needs to be serialized to go into local storage)